src/drive/web/modules/drive/FileOpenerExternal.jsx

```diff
@@ -27,16 +27,27 @@ class FileOpener extends Component {
     file: null
   }
   componentWillMount() {
-    this.loadFileInfo()
+    if (this.props.fileId) {
+      this.loadFileInfo(this.props.fileId)
+    } else {
+      if (this.props.routeParams.fileId) {
+        this.loadFileInfo(this.props.routeParams.fileId)
+      }
+    }
   }
 
-  async loadFileInfo() {
+  componentWillReceiveProps(nextProps) {
```

Comment:
I would prefer that we don't use `componentWillReceiveProps` ( https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops ). We should use `getDerivedStateFromProps` for this (or maybe `componentDidUpdate` as we are performing a side effect).

By the way, why do you need to do this in both `componentWillMount` and `componentWillReceiveProps` ?

Reply:
Doing it in `componentWillReceiveProps ` solves the bug that we had, because the component is not unmounted in some cases, but the document changes.

But as you say it should be possible to use something else than `componentWillReceiveProps` here :)

Reply:
> I would prefer that we don't use componentWillReceiveProps ( https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops ). We should use getDerivedStateFromProps for this (or maybe componentDidUpdate as we are performing a side effect).

Because we're not using React yet... 

> By the way, why do you need to do this in both componentWillMount and componentWillReceiveProps ?

Because `componentWillReceiveProps` is not call during the initial render()

Reply:
OK, `getDerivatedStateFromProps` was added to Preact 8.3.0, the version we use. 

But DidUdapte will do the job :)

Reply:
Resolved. I use Didupdate, but again, I need to call it in the constructor and in the didupdate method ;)